### PR TITLE
fix(FEC-14000): Decoupling Plugins from Audio Player

### PR DESCRIPTION
**Pull request by JonathanTGold**
Decoupling Plugins from Audio Player

Related pr:
https://github.com/kaltura/kaltura-player-js/pull/805
https://github.com/kaltura/playkit-js-audio-player/pull/27
https://github.com/kaltura/playkit-js-downloads/pull/56
https://github.com/kaltura/playkit-js-share/pull/50

### CODE REVIEW

File: src/transcript-plugin.tsx [hunk 1]
```diff
@@ -393,14 +399,18 @@ export class TranscriptPlugin extends KalturaPlayer.core.BasePlugin {
     this._pluginState = PluginStates.CLOSED;
   };
 
-  public isPluginAvailable = () => {
+  public isEntrySupported(): boolean {
```
Comment (JonathanTGold) [inline src/transcript-plugin.tsx:402]: I think isPluginAvailableForEntry is better descriptive name

File: src/transcript-plugin.tsx [hunk 2]
```diff
@@ -410,8 +419,11 @@ export class TranscriptPlugin extends KalturaPlayer.core.BasePlugin {
     if (Math.max(this._transcriptPanel, this._transcriptIcon) > 0) {
       this.sidePanelsManager?.remove(this._transcriptPanel);
       this.upperBarManager!.remove(this._transcriptIcon);
+      // @ts-ignore
+      this.player.getService('AudioPluginsManager').remove(this._audioPlayerIconId);
```
Comment (SivanA-Kaltura) [inline src/transcript-plugin.tsx:423]: you can add a getter function for the service

File: src/transcript-plugin.tsx [hunk 3]
```diff
@@ -271,11 +278,10 @@ export class TranscriptPlugin extends KalturaPlayer.core.BasePlugin {
   };
 
   private _addTranscriptItem(): void {
-    if (Math.max(this._transcriptPanel, this._transcriptIcon) > 0) {
+    if (Math.max(this._transcriptPanel, this._transcriptIcon, this._audioPlayerIconId) > 0) {
```
Comment (SivanA-Kaltura) [inline src/transcript-plugin.tsx:281]: why is this needed ?

**Review by SivanA-Kaltura — CHANGES_REQUESTED** [review]

**Review by semarche-kaltura — APPROVED** [review]

File: src/transcript-plugin.tsx (hunk 2 above)
Comment (JonathanTGold) [reply src/transcript-plugin.tsx:423]: we need to remove the former not do the later
